tests/test_auth.py

```diff
@@ -89,81 +84,34 @@ def test_invalid_token(self, client, monkeypatch):
 
 
 class TestOIDCAuthentication(object):
-    invalid_token_error = "Token required but invalid"
-    auth_missing_error = "No 'Authorization' header found"
-
-    def test_get_user_without_token(self, session):
-        with pytest.raises(Unauthorized) as excinfo:
-            request = mock.MagicMock()
-            waiverdb.auth.get_user(request)
+    auth_missing_error = "401 Unauthorized: Failed to retrieve username"
+
+    def test_get_user_no_auth_methods(self):
+        with mock.patch('waiverdb.auth.auth_methods') as mocked:
+            mocked.return_value = []
+            with pytest.raises(Unauthorized) as excinfo:
+                waiverdb.auth.get_user(request)
+        assert "Authenticated user required. No methods specified." in str(excinfo.value)
+
+    def test_get_user_without_token(self, app):
+        with app.test_request_context('/api/v1.0/waivers/new'):
+            with pytest.raises(Unauthorized) as excinfo:
+                waiverdb.auth.get_user(request)
         assert self.auth_missing_error in str(excinfo.value)
 
-    def test_get_user_with_invalid_token(self, oidc_token, session):
```

Comment:
Why were the test functions removed? I think mainly the `test_warning_banner` decreases code covered by tests (`_authorization_warning_from_exception()` and `_authorization_warning()` are not covered).

Reply:
Remade tests for `_authorization_warning` and created for `verify_authorization`. Coverage is ok now.

Reply:
Cool. What about testing behavior on invalid/expired token?

Reply:
Checking of the token is now the libraries duty. I don't think we should test that the libraries are working correctly.